packages/react-components/react-toolbar/src/components/ToolbarToggleButton/ToolbarToggleButton.tsx

```diff
@@ -12,9 +12,23 @@ import { useToolbarContext } from '../Toolbar/ToolbarContext';
  * ToolbarToggleButton component
  */
 export const ToolbarToggleButton: ForwardRefComponent<ToolbarToggleButtonProps> = React.forwardRef((props, ref) => {
-  const { size } = useToolbarContext();
-
+  const { size, handleToggleButton } = useToolbarContext();
+  const { onClick: onClickOriginal } = props;
   const state = useToggleButton_unstable({ size, ...props }, ref);
+  const handleOnClick = (
+    e: React.MouseEvent<HTMLButtonElement, MouseEvent> & React.MouseEvent<HTMLAnchorElement, MouseEvent>,
+  ) => {
+    if (state.disabled) {
+      e.preventDefault();
+      e.stopPropagation();
+      return;
+    }
+
+    handleToggleButton?.(e, state.name, state.value, state.checked);
+    onClickOriginal?.(e);
+  };
+
+  state.root.onClick = handleOnClick;
 
   useToggleButtonStyles_unstable(state);
   return renderToggleButton_unstable(state);
```

Comment:
How about we keep following the compositional API? we need to have `useComponent`, `useComponentStyles` and `renderComponent` functions separately implemented, even though they're redundant and just call `useToggleButtonStyles_unstable` 😅

Reply:
Oh, the decision to not have them was made in some of the Fluent syncs as way to reuse components we already have.

Reply:
AFAIR there was a discussion indeed and the conclusion was not to create new hooks if they do **nothing** as the whole API unstable. In that PR we have some logic in a component, so IMO there should be at least `useToolabarToggleButton_unstable`.

Reply:
Yep, we have a `useToolabarToggleButton_unstable`